xgcm/grid.py

```diff
@@ -1235,7 +1235,10 @@ def add_to_slice(da, dim, sl, value):
     # beginning or end before or after will be empty)
     before = da[{dim: slice(0, sl)}]
     middle = da[{dim: sl}]
+    # the newest xarray version seems to not retain the singular dimension
+    middle = middle.expand_dims(dim)
```

Comment:
it seems like the newest version of xarray does not retain a singular dimension by default. This fix restores that behaviour.

Reply:
@dcherians fix over at xarray will fix this, but our tests will still fail until a new version of xarray is released, or am I missing something here? Should we leave this patch in and mark it for removal in the future?

Reply:
Update the comment with a reference to the xarray PR and version with which it will become unnecessary.

It's fine to leave it in if it has no performance penalty.